scipy/fft/tests/test_basic.py

```diff
@@ -354,11 +482,75 @@ def test_multiprocess(func):
 
 class TestIRFFTN:
 
-    def test_not_last_axis_success(self):
+    @array_api_compatible
+    @skip_if_array_api_backend('torch')
+    def test_not_last_axis_success(self, xp):
         ar, ai = np.random.random((2, 16, 8, 32))
         a = ar + 1j*ai
+        a = xp.asarray(a)
 
         axes = (-2,)
 
         # Should not raise error
         fft.irfftn(a, axes=axes)
+
+
+class TestNamespaces:
+
+    @array_api_compatible
+    @pytest.mark.parametrize("func", [fft.fft, fft.ifft])
+    def test_fft_ifft(self, func, xp):
+        x = xp.asarray(random(30) + 1j*random(30))
+        _assert_matching_namespace(func(x), x)
+
+    # torch.fft not yet implemented by array-api-compat
+    @skip_if_array_api_backend('torch')
+    @array_api_compatible
+    @pytest.mark.parametrize("func", [fft.fftn, fft.ifftn])
+    def test_fftn_ifftn(self, func, xp):
+        x = xp.asarray(random((30, 20, 10)) + 1j*random((30, 20, 10)))
+        _assert_matching_namespace(func(x), x)
+
+    @array_api_compatible
+    def test_rfft(self, xp):
+        x = xp.asarray(random(29))
+        _assert_matching_namespace(fft.rfft(x), x)
+
+    @array_api_compatible
+    def test_irfft(self, xp):
+        x = xp.asarray(random(30))
+        _assert_matching_namespace(fft.irfft(x), x)
+
+    # torch.fft not yet implemented by array-api-compat
+    @skip_if_array_api_backend('torch')
+    @array_api_compatible
+    @pytest.mark.parametrize("func", [fft.rfftn, fft.irfftn])
+    def test_rfftn_irfftn(self, func, xp):
+        x = xp.asarray(random((30, 20, 10)))
+        _assert_matching_namespace(func(x), x)
+
+    @array_api_compatible
+    def test_hfft_ihfft(self, xp):
+        x = random(14) + 1j*random(14)
+        x_herm = np.concatenate((random(1), x, random(1)))
+        x_herm = xp.asarray(x_herm)
+        y = fft.hfft(x_herm)
+        _assert_matching_namespace(y, x_herm)
+        _assert_matching_namespace(fft.ihfft(y), y)
+
+
+# torch.fft not yet implemented by array-api-compat
+@skip_if_array_api_backend('torch')
```

Comment:
Is this just a testing infra issue?

Reply:
>  I don't think that we should have moved from keyword args to positional args

Can't you still get away with it via `**kwargs` instead of restoring the code duplication? Also, part of my original point was that the tests still passed--if this is really important a test should perhaps fail if we don't conform to it.

To be clear though, I was never convinced that SciPy *itself* should be a *provider* of the standard, just that it would support it *internally* to allow swapping backends. I think I made a similar comment for `linalg` discussion re: us being an algorithms library vs. array API provider library. I can understand why one might want to re: mirroring NumPy swaps perhaps, or if it reduces confusion.

Reply:
> To be clear though, I was never convinced that SciPy itself should be a provider of the standard, just that it would support it internally to allow swapping backends.

Yes I agree, this is just about whether `torch` is a provider.

> Can't you still get away with it via **kwargs instead of restoring the code duplication?

Ah possibly, I'll try that 👍

Reply:
Ah so PyTorch allows `dim` arguments to be called with `axis` but not  `axes`. It is a bit annoying that NumPy makes this distinction, especially as it isn't inconsistent. e.g. `np.squeeze(x, axis=(1, 2))`.

Reply:
@tylerjereddy is this what you had in mind? https://github.com/scipy/scipy/pull/19005/commits/5f68e8ce73e231288a4ead590fe743c8a1a576ff

Reply:
We can leave this to a follow-up if needed, but to me it looks like the cure is worse than the disease here. The use of `**kwargs` and multiple `try`-`except` clauses to avoid a bit of code duplicate between `_execute` and `_execute_nD` feels a bit off. @tylerjereddy WDYT?

Reply:
The new solution is probably still less lines, but I'm happy for you and Lucas to proceed with a revert-to-original if you prefer the original case.

Reply:
No strong preference from me. If I had to say, I'd go for the separate functions to avoid the `try`-`except`s. Your call @rgommers.

Reply:
I think the original is preferred - easier to read and more robust. @lucascolley can you please take this along in a follow-up PR?